src/middlewares/logger.middleware.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable, Logger, NestMiddleware } from '@nestjs/common';
+import { Request, Response, NextFunction } from 'express';
+import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class LoggerMiddleware implements NestMiddleware {
+  private readonly logger = new Logger();
+  private readonly jwt = new JwtService();
+
+  use(req: Request, res: Response, next: NextFunction) {
+    const token = req.headers?.authorization?.split(' ')[1] || '';
+    const userDetail = this.jwt.decode(token) || {};
```

Comment:
PullProofAI

The code assumes that the authorization header is always present and in the format "Bearer <token>". It's recommended to add proper error handling for cases where the header is missing or not in the expected format.

```diff
+     const token = req.headers?.authorization?.split(' ')[1] || '';
+     const userDetail = this.jwt.decode(token) || {};
```



<!-- This is an auto-generated comment by PullProofAI -->